students/shawn_degraw/session02/series.py

```diff
@@ -0,0 +1,72 @@
+def fibonacci(n):
+    """
+    compute the nth Fibonacci number 
+    
+    :param n: value of the nth fibonacci element
+    """
+    return sum_series(n)
+
+
+def lucas(n):
+    """
+    compute the nth Lucas number
+    
+    :param n: value of the nth lucas element
+    """
+    return sum_series(n, 2, 1)
+
+def sum_series(n, n0=0, n1=1):
+    """
+    compute the nth value of a summation series.
+
+    :param n0=0: value of zeroth element in the series
+    :param n1=1: value of first element in the series
+
+    if n0 == 0 and n1 == 1, the result is the Fibbonacci series
+
+    if n0 == 2 and n1 == 1, the result is the Lucas series
+
+    if n0 and n1 not matching, series not implemented yet
+    """
+
+    """ compute the nth Fibonacci number """
+    if n0 == 0 and n1 == 1:
+        if n == 0:
+            return 0
+        elif n == 1:
+            return 1
+        else:
+            return sum_series(n-1) + sum_series(n-2)
+    """ compute the nth Lucas number """
+    if n0 == 2 and n1 == 1:
+        if n == 0:
+            return 2
+        elif n == 1:
+            return 1
+        else:
+            return sum_series(n-1, 2, 1) + sum_series(n-2, 2, 1)
+    """ Other series: not implemented yet """
+    return None
```

Comment:
So, this was left not done? I was expecting this to be completed, leading to the generalization of the above two cases. Geez, those comments in the template did seem to have confused many students...